server/src/main/java/org/opensearch/index/remote/RemoteTranslogTracker.java

```diff
@@ -0,0 +1,481 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.remote;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.common.util.MovingAverage;
+import org.opensearch.core.index.shard.ShardId;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Stores Remote Translog Store-related stats for a given IndexShard.
+ */
+public class RemoteTranslogTracker {
```

Comment:
can we extend `TranslogStats` for remote instead of using pressure service ? Since we are not doing backpressure from translog , we don't need to integrate with it . Translog stats give nice integration with node stats API .

Reply:
Makes sense, but is it confirmed that we won't be introducing backpressure for Remote Translog Store uploads? I understand it may be more relevant for async translog durability

Reply:
@gbbafna resolving this thread as we are surfacing the RTS upload stats under nodes stats API too now
Thanks!